src/services/webdav/backend.rs

```diff
@@ -172,9 +200,18 @@ impl Builder for WebdavBuilder {
             })?
         };
 
+        let auth = match(&self.username, &self.password) {
+            (Some(username), Some(password)) => {
+                // base64 encode
+                general_purpose::STANDARD.encode(format!("{}:{}", username, password))
+            },
+            _ => { String::default() }
```

Comment:
`None, Some(password)` means that the user is misconfigured. It is better to return an invalid error to notice our user.